youtube_dl/extractor/infomaniak.py

```diff
@@ -0,0 +1,88 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    parse_duration,
+)
+
+
+class InfomaniakVod2IE(InfoExtractor):
+    IE_NAME = 'infomaniak:vod2'
+    _VALID_URL = r'https?://player\.vod2\.infomaniak\.com/embed/(?P<id>[0-9a-z]+)'
+    _TEST = {
+        'url': 'https://player.vod2.infomaniak.com/embed/1jhvl2uqg6ywp',
+        'info_dict': {
+            'id': '1jhvl2uqg6xis',
+            'display_id': '1jhvl2uqg6ywp',
+            'ext': 'mp4',
+            'title': 'Conférence à Dyo, octobre 2022',
+            'thumbnail': r're:^https?://.*\.(?:jpe?g|png)$',
+        },
+        'params': {
+            'skip_download': True,
+        },
+    }
+
+    def _real_extract(self, url):
+        share_id = self._match_id(url)
+        share = self._download_json(
+            'https://api.infomaniak.com/2/vod/res/shares/%s.json' % share_id,
+            share_id)
+        data = (share or {}).get('data') or {}
+        media = data.get('media') or []
+
+        entries = []
+        for m in media:
+            if not isinstance(m, dict):
+                continue
```

Comment:
More robust and concise, since `data` and `media` aren't being used elsewhere,
```suggestion
        entries = []
        for m in traverse_obj(share, (
                'data', 'media', lambda _, v: v.get('source').get)):
            media_id = m.get('id') or share_id
```
Then `m`, not a great variable name, could be renamed as `media`.

Reply:
Agreed. this is more cleaner. I refactored the parsing to use traverse_obj(...) for iterating media entries and renamed the loop variable to media for readability.